### PR TITLE
detect: update packet action on protocol change

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6305

Describe changes:
- detect: update packet action on protocol change

#9541 rebased and not a draft

### CODE REVIEW

**Comment by victorjulien** [comment]
Do you have a SV test for this?

File: src/flow-worker.c [hunk 1]
```diff
@@ -430,6 +430,10 @@ static inline void FlowWorkerStreamTCPUpdate(ThreadVars *tv, FlowWorkerThreadDat
             TmqhOutputPacketpool(tv, x);
         }
     }
+    if (FlowChangeProto(p->flow) && p->flow->flags & FLOW_ACTION_DROP) {
+        // in case f->flags & FLOW_ACTION_DROP was set by one of the dequeued packets
+        PacketDrop(p, ACTION_DROP, PKT_DROP_REASON_FLOW_DROP);
```
Comment (victorjulien) [inline src/flow-worker.c:435]: this drop reason isn't correct, it should pass on  the original reason given by the tunnel packet

**Comment by catenacyber** [comment]
> Do you have a SV test for this?

Adding one in https://github.com/OISF/suricata-verify/pull/1619

File: src/flow-worker.c (hunk 1 above)
Comment (catenacyber) [reply src/flow-worker.c:435]: What do you call a "tunnel" packet ? There is no GRE or over tunneling protocols here...

Why is not the reason correct ?
`FlowHandlePacketUpdate` does
```
    if (f->flags & FLOW_ACTION_DROP) {
        PacketDrop(p, ACTION_DROP, PKT_DROP_REASON_FLOW_DROP);
    }
```

same reason for same condition : flow flags have `FLOW_ACTION_DROP` 

so I do not put reason `PKT_DROP_REASON_RULES`

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/10270

**Pull request closed without merging**